intermediate_notebooks/E2E/mortgage/mortgage_e2e.ipynb

```diff
@@ -0,0 +1,772 @@
+    "\n",
+    "    dmat = xgb.dask.DaskDMatrix(\n",
+    "        client=client,\n",
+    "        data=persisted_train_x_y[\n",
+    "            persisted_train_x_y.columns.difference([\"delinquency_12\"])\n",
+    "        ],\n",
+    "        label=persisted_train_x_y[[\"delinquency_12\"]],\n",
+    "        missing=-1,\n",
+    "    )\n",
+    "\n",
+    "    del persisted_train_x_y\n",
+    "    gc.collect()\n",
+    "\n",
+    "    dmat_time = time.time() - t1\n",
+    "    print(\"Prepared data for XGB training\")\n",
+    "\n",
+    "    print(\"Training model\")\n",
+    "    t1 = time.time()\n",
+    "\n",
+    "    print(\"XGB training for part_count:{}\".format(part_count))\n",
+    "    bst = xgb.dask.train(\n",
+    "        client, dxgb_gpu_params, dmat, num_boost_round=NUM_BOOST_ROUND,\n",
+    "    )\n",
+    "\n",
+    "    train_time = time.time() - t1\n",
+    "    print(\"Training complete\")\n",
+    "    return (bst, dmat_time, train_time)\n",
+    "\n",
+    "\n",
+    "def run_etl(start_year, end_year, data_dir, client):\n",
+    "\n",
+    "    print(\"Starting ETL\")\n",
+    "    t1 = time.time()\n",
+    "\n",
+    "    perf_data_path = data_dir + \"perf/\"\n",
+    "\n",
+    "    gpu_dfs = []\n",
+    "    quarter = 1\n",
+    "    year = start_year\n",
+    "    count = 0\n",
+    "    while year <= end_year:\n",
+    "        for file in glob(\n",
+    "            os.path.join(\n",
+    "                perf_data_path + \"/Performance_\" + str(year) + \"Q\" + str(quarter) + \"*\"\n",
+    "            )\n",
+    "        ):\n",
+    "            gpu_dfs.append(\n",
+    "                process_quarter_gpu(\n",
+    "                    year=year, quarter=quarter, perf_file=file, client=client\n",
+    "                )\n",
+    "            )\n",
+    "            count += 1\n",
+    "        quarter += 1\n",
+    "        if quarter == 5:\n",
+    "            year += 1\n",
+    "            quarter = 1\n",
+    "    print(\"ETL for start_year:{} and end_year:{}\".format(start_year, end_year))\n",
+    "    wait(gpu_dfs)\n",
+    "\n",
+    "    etl_time = time.time() - t1\n",
+    "\n",
+    "    print(\"ETL done!\")\n",
+    "    return (gpu_dfs, etl_time)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### The cell below runs the workflow end to end including the ETL and XGBoost\n",
+    "\n",
+    "**Note:** The mortgage dataset for years 2000-2016 is about 200GB of data. The total GPU memory and the max memory per gpu are two key parameters that determine the amount of data the system would be able to process for the workflow. The `determine_dataset` utility used below automatically queries these two parameters on the machine running the script and decides suitable values `start_year`, `end_year`, `part_count` and variation of the dataset (1GB split part files vs 2GB split part files) that should work on such systems.\n",
```

Comment:
> determine the amount of data the system would be able to process for the workflow

It seems like we should clarify that the ETL can run for all years, but that XGBoost needs to load the entire training set into GPU memory, and thus part count is the real limiting factor.